08_education/digital_access.qmd

```diff
@@ -0,0 +1,939 @@
+      )
+  }
+
+  data |>
+    arrange(year, state, !!geo_var, state_name, paste0(geo_var, "_name"))
+}
+
+
+lst(
+  county_overall_digital_access,
+  place_overall_digital_access
+) %>%
+  map2(
+    .x = names(.),
+    .y = .,
+    .f = ~ assign(
+      x = .x,
+      value = reshape_for_export(.y, subgroup = "overall"),
+      envir = .GlobalEnv
+    )
+  )
+
+lst(
+  county_raceeth_digital_access,
+  place_raceeth_digital_access
+) %>%
+  map2(
+    .x = names(.),
+    .y = .,
+    .f = ~ assign(
+      x = .x,
+      value = reshape_for_export(.y, subgroup = "race-ethnicity"),
+      envir = .GlobalEnv
+    )
+  )
+
+# 3144 obs (counties) * 5 race groups = 15,720 obs -- accurate
+count(county_raceeth_digital_access, year)
+
+# 486 obs (places) * 5 race groups = 2,430 obs -- accurate
+count(place_raceeth_digital_access, year)
+
+lst(
+  county_income_digital_access,
+  place_income_digital_access
+) %>%
+  map2(
+    .x = names(.),
+    .y = .,
+    .f = ~ assign(
+      x = .x,
+      value = reshape_for_export(.y, subgroup = "income"),
+      envir = .GlobalEnv
+    )
+  )
+```
+
+We suppress showing metric values for observations with missing CVs
+
+```{r}
+#| label: suppress-bad-cvs
+
+suppress_bad_cvs <- function(data) {
+  data %>%
+    mutate(across(
+      c(share_digital_access, share_digital_access_lb, share_digital_access_ub),
+      ~ if_else(is.na(share_digital_access_quality), NA_real_, .x)
+    ))
+}
+
+lst(
+  county_overall_digital_access,
+  place_overall_digital_access,
+  county_raceeth_digital_access,
+  place_raceeth_digital_access,
+  county_income_digital_access,
+  place_income_digital_access
+) %>%
+  map2(
+    .x = names(.),
+    .y = .,
+    .f = ~ assign(
+      x = .x,
+      value = suppress_bad_cvs(.y),
+      envir = .GlobalEnv
+    )
+  )
+
+county_subgroup_digital_access <-
+  bind_rows(
+    county_raceeth_digital_access,
+    county_income_digital_access
+  )
+
+place_subgroup_digital_access <-
+  bind_rows(
+    place_raceeth_digital_access,
+    place_income_digital_access
+  )
+```
+
+```{r}
+get_summary_stats <- function(data) {
+  data |>
+    summarise(across(c(share_digital_access, share_digital_access_lb, share_digital_access_ub),
+      .fns = list(
+        min = ~ min(.x, na.rm = TRUE),
+        median = ~ median(.x, na.rm = TRUE),
+        max = ~ max(.x, na.rm = TRUE)
+      ),
+      .names = "{.col}_{.fn}"
+    )) |>
+    pivot_longer(cols = everything()) |>
+    mutate(
+      stat = str_extract(name, "[^_]+$"),
+      name = str_remove(name, "_[^_]+$")
+    ) |>
+    pivot_wider(
+      names_from = stat,
+      values_from = value
+    )
+}
+
+county_overall_digital_access |> get_summary_stats()
+
+place_overall_digital_access |> get_summary_stats()
+
+county_raceeth_digital_access |> get_summary_stats()
+
+place_raceeth_digital_access |> get_summary_stats()
+
+county_income_digital_access |> get_summary_stats()
+
+place_income_digital_access |> get_summary_stats()
+```
```

Comment:
could we add a comment here re: how many get suppressed when we do this?

Reply:
I found that no observations were actually supressed because there aren't any missing quality flag values when the metric
value is not missing! Added a comment saying this.